File: backend/app/schemas/emergency_resource.py
from app.schemas.common import DatetimeStr
from typing import Optional
from pydantic import BaseModel

class EmergencyResourceCreate(BaseModel):
    category: str; name: str; specification: str | None = None; quantity: int = 0
    unit: str | None = None; location: str | None = None; responsible_person: str | None = None
    contact_phone: str | None = None; is_external: bool = False
    external_address: str | None = None; external_distance_km: float | None = None

class EmergencyResourceUpdate(BaseModel):
    category: str | None = None; name: str | None = None; specification: str | None = None
    quantity: int | None = None; unit: str | None = None; location: str | None = None
    responsible_person: str | None = None; contact_phone: str | None = None
    is_external: bool | None = None; external_address: str | None = None
    external_distance_km: float | None = None

class EmergencyResourceResponse(BaseModel):
    id: str; enterprise_id: str; category: str; name: str; specification: str | None
    quantity: int; unit: str | None; location: str | None; responsible_person: str | None
    contact_phone: str | None; is_external: bool; external_address: str | None
    external_distance_km: float | None; created_at: DatetimeStr
    model_config = {"from_attributes": True}

File: frontend/src/types/emergencyResource.ts
export interface EmergencyResource {
  id: string;
  enterprise_id: string;
  category: string;
  name: string;
  specification: string;
  quantity: number;
  unit: string;
  location: string;
  responsible_person: string;
  contact_phone: string;
  is_external: boolean;
  external_address: string;
  external_distance_km: number | null;
  created_at: string;
}

export interface EmergencyResourceCreate {
  category: string;
  name: string;
  specification?: string;
  quantity?: number;
  unit?: string;
  location?: string;
  responsible_person?: string;
  contact_phone?: string;
  is_external?: boolean;
  external_address?: string;
  external_distance_km?: number | null;
}

export interface EmergencyResourceUpdate extends Partial<EmergencyResourceCreate> {}
